docs/lang.rst

```diff
@@ -235,6 +235,9 @@ Casting a value from an integer ``i`` is a shorthand for ``Const(i)``:
    >>> Value.cast(5)
    (const 3'd5)
 
+.. note::
+
+   If a value subclasses :class:`enum.IntEnum` or is otherwise both int and Enum, Enum behavior takes precedence.
```

Comment:
I'd phrase this as:

    If a value subclasses :class:`enum.IntEnum` or its class otherwise inherits from both :class:`int` and :class:`enum.Enum`, it is treated as an enumeration.